File: atcoder/python/abc245/c.py
n,k=map(int,input().split())
a=list(map(int,input().split()))
b=list(map(int,input().split()))
dp=[[0]*2 for i in range(n)]
dp[0][0]=dp[0][1]=1
for i in range(1,n):
    if abs(a[i]-a[i-1])<=k:
        dp[i][0]|=dp[i-1][0]
    if abs(a[i]-b[i-1])<=k:
        dp[i][0]|=dp[i-1][1]
    if abs(b[i]-a[i-1])<=k:
        dp[i][1]|=dp[i-1][0]
    if abs(b[i]-b[i-1])<=k:
        dp[i][1]|=dp[i-1][1]
if dp[n-1][0]>0 or dp[n-1][1]>0:
    print("Yes")
else:
    print("No")

File: atcoder/java/abc245/c.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        try(Scanner sc = new Scanner(System.in)) {
            int n = sc.nextInt();
            int k = sc.nextInt();
            int[] a = new int[n];
            for(int i=0; i<n; i++) {
                a[i] = sc.nextInt();
            }
            int[] b = new int[n];
            for(int i=0; i<n; i++) {
                b[i] = sc.nextInt();
            }
            var dp = new boolean[n];
            var ep = new boolean[n];
            dp[0] = true;
            ep[0] = true;
            for(int i=1; i<n; i++) {
                if(dp[i-1] && Math.abs(a[i-1]-a[i]) <= k) {
                    dp[i] = true;
                } else if(ep[i-1] && Math.abs(b[i-1]-a[i]) <= k) {
                    dp[i] = true;
                }
                if(dp[i-1] && Math.abs(a[i-1]-b[i]) <= k) {
                    ep[i] = true;
                } else if(ep[i-1] && Math.abs(b[i-1]-b[i]) <= k) {
                    ep[i] = true;
                }
            }
            if(dp[n-1] || ep[n-1]) {
                System.out.println("Yes");
            } else {
                System.out.println("No");
            }
        }
    }
}